include/mata/nfa/nfa.hh

```diff
@@ -226,6 +226,19 @@ public:
      */
     Nfa& trim(StateRenaming* state_renaming = nullptr) { return trim_inplace(state_renaming); }
 
+    /**
+     * @brief Remove inaccessible (unreachable) and not co-accessible (non-terminating) states from a copy of the
+     *  original automaton.
+     *
+     * Remove states which are not accessible (unreachable; state is accessible when the state is the endpoint of a path
+     *  starting from an initial state) or not co-accessible (non-terminating; state is co-accessible when the state is
+     *  the starting point of a path ending in a final state).
+     *
+     * @param[out] state_renaming Mapping of trimmed states to new states.
+     * @return New @c Nfa instace from @c this with trimmed states.
+     */
+    Nfa trimmed(StateRenaming* state_renaming = nullptr) const { return Nfa{ *this }.trim(state_renaming); }
```

Comment:
I think this naming is confusing, having `trim` and `trimmed` will only confuse users, so I suggest to brainstorm some better name.

Reply:
The idea is that the operations that work in-place will usually have the name after the operation (usually a verb) and what it does to the instance the operation is called on. The operations returning a new instance will be named after the result, the new instance, which in this case is the trimmed NFA. Again, this is the idea. Let us discuss this here. `trim()` and `trimmed()` are the example of how it will look like. If we agree on using this pattern, all other functions will be renamed similarly.

Reply:
I get the idea, but the naming is probably not ideal. I though of `copy_trimmed()` or `clone_trimmed()` to show that a copy of the original automaton is made.

I think, we should unify naming convetion (mainly) for: 1. member operations (like `nfa.trim()`, 2. non-member operation (like `res = trim(nfa)`, 3. member operation returning clone/copy `res = nfa.trim()`.

Reply:
>  I though of copy_trimmed() or clone_trimmed() to show that a copy of the original automaton is made.

I would prefer something like this, but others have to agree first, too.

>  unify naming convetion

That is what I start doing here. This is an example of how it will look with `trim()`, other function will follow.

Reply:
That can be done. I will remove `trimmed()` for now.